api.planx.uk/shared/middleware/validate.ts

```diff
@@ -0,0 +1,39 @@
+import { Request, RequestHandler, Response, NextFunction } from "express";
+import { AnyZodObject, ZodSchema, z } from "zod";
+
+/**
+ * Middleware to validate incoming requests to the API
+ * Takes a ZodSchema and returns a validated, and typed, Request object
+ */
+export const validate =
+  (schema: AnyZodObject) =>
+  async (
+    req: Request<z.infer<typeof schema>>,
```

Comment:
I think this may not be _perfectly_ type safe as it's not using a generic, but it's safe enough that we can trust it when we type out Express request handlers with `ValidatedRequestHandler<YourSchema, YourResponse>`👌 

I spent a bit of time trying to get this to use this type and it ended up being a bit of a rabbit hole so I'm happy with this for now.